registry/consul/options.go

```diff
@@ -0,0 +1,25 @@
+package consul
+
+import (
+	consul "github.com/hashicorp/consul/api"
+	"github.com/micro/go-micro/registry"
+	"golang.org/x/net/context"
+)
+
+func Config(c *consul.Config) registry.Option {
+	return func(o *registry.Options) {
+		if o.Context == nil {
+			o.Context = context.Background()
+		}
+		o.Context = context.WithValue(o.Context, "consul_config", c)
+	}
+}
+
+func Token(t string) registry.Option {
+	return func(o *registry.Options) {
+		if o.Context == nil {
+			o.Context = context.Background()
+		}
+		o.Context = context.WithValue(o.Context, "consul_token", t)
```

Comment:
Would it be worth appending that to a default config and then use the `Config` method as above? This way you only have 1 key that does this work.

On the key part, would it be feasible to add a `NewContext` and `FromContext` method (like with metadata) which sets/gets the config from a private key (Have a look [here](https://github.com/micro/go-micro/blob/master/metadata/metadata.go#L7))?